src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/custom.py

```diff
@@ -582,6 +629,95 @@ def set_diagnostics_extension(
+                                          type=vm_extension_name,
+                                          protected_settings=private_config,
+                                          type_handler_version=version,
+                                          settings=settings,
+                                          auto_upgrade_minor_version=auto_upgrade)
+
+    vmss.virtual_machine_profile.extension_profile.extensions.append(ext)
+
+    poller = client.virtual_machine_scale_sets.create_or_update(resource_group_name,
+                                                                vmss_name,
+                                                                vmss)
+    return ExtensionUpdateLongRunningOperation('updating diagnostics', 'done')(poller)
+
+def get_vmss_extension(resource_group_name, vmss_name, extension_name):
+    client = _compute_client_factory()
+    vmss = client.virtual_machine_scale_sets.get(resource_group_name,
+                                                 vmss_name)
+    #pylint: disable=no-member
+    if not vmss.virtual_machine_profile.extension_profile:
+        return
+    return next((e for e in vmss.virtual_machine_profile.extension_profile.extensions
+                 if e.name == extension_name), None)
+
+def list_vmss_extensions(resource_group_name, vmss_name):
+    client = _compute_client_factory()
+    vmss = client.virtual_machine_scale_sets.get(resource_group_name,
+                                                 vmss_name)
+    #pylint: disable=no-member
+    return None if not vmss.virtual_machine_profile.extension_profile \
+        else vmss.virtual_machine_profile.extension_profile.extensions
+
+def delete_vmss_extension(resource_group_name, vmss_name, extension_name):
+    client = _compute_client_factory()
+    vmss = client.virtual_machine_scale_sets.get(resource_group_name,
+                                                 vmss_name)
+    #pylint: disable=no-member
+    if not vmss.virtual_machine_profile.extension_profile:
+        raise CLIError('Scale set has no extensions to delete')
+
+    keep_list = [e for e in vmss.virtual_machine_profile.extension_profile.extensions
+                 if e.name != extension_name]
+    if len(keep_list) == len(vmss.virtual_machine_profile.extension_profile.extensions):
+        raise CLIError('Extension {} not found'.format(extension_name))
+
+    vmss.virtual_machine_profile.extension_profile.extensions = keep_list
+
+    return client.virtual_machine_scale_sets.create_or_update(resource_group_name,
+                                                              vmss_name,
+                                                              vmss)
+
+def _get_private_config(resource_group_name, storage_account):
+    storage_mgmt_client = _get_storage_management_client()
+    #pylint: disable=no-member
+    keys = storage_mgmt_client.storage_accounts.list_keys(resource_group_name, storage_account).keys
```

Comment:
>keys = storage_mgmt_client.storage_accounts.list_keys(resource_group_name, storage_account).keys [](start = 4, length = 96)

So is this going to be one of those cross-cutting versioning headaches that we alluded to last week?